### PR TITLE
Developer workflow improvements

### PR DESCRIPTION
**Pull request by mrfelton**
Following on from the work in https://github.com/LN-Zap/zap-desktop/pull/439 I've made several additional improvements to the developer workflow here.

The goal of this PR is to:
- provide consistency in development standards
- automate common development workflow tasks
- provide additional help for new contributors

The updates include:
 - Integration with [`prettier`](https://github.com/prettier/prettier) via [`prettier-eslint`](https://github.com/prettier/prettier-eslint)
- Automatic code formatting and linting in git `pre-commit` hook
- Automatic commit message format validation in git `commit-msg` hook
- Above commit hooks are opt-in (enable them by creating a `.opt-in` file (see updated docs)
- Multiple updates to documentation:

    **Readme (see https://github.com/mrfelton/zap-desktop/blob/build/developer-workflow/README.md)**
     - Update format to conform to `standard-readme` format
     - Tidy up header section
     - Add table of contents
     - Add security section
     - Add help section
     - List maintainers
     - Add contribution notes
     - Add contributors list

    **Contributing (see https://github.com/mrfelton/zap-desktop/blob/build/developer-workflow/CONTRIBUTING.md)**
     - Add table of contents
     - Add contribution faq
     - Add contribution guidelines

     **Advanced (see https://github.com/mrfelton/zap-desktop/blob/build/developer-workflow/ADVANCED.md)**
      - Add table of contents

### CODE REVIEW

**Comment by mrfelton** [comment]
One of the really cool things about Prettier is that it has good integration with a large number of text editors. For example, with Atom if you have https://github.com/prettier/prettier-atom installed it will run Prettier whenever you save a file - so you literally don't have to worry at all about things like spacing, line length, semicolons, single vs double quotes as it will auto correct everything whenever you save the file.

See https://prettier.io/

File: README.md [hunk 1]
```diff
@@ -72,22 +90,51 @@ Once you have the .AppImage file extracted, you can either **double click** the
 ```
 
 ## Advanced Usage
-If you would like to install from source or run a full bitcoin node, please see the [advanced usage](https://github.com/LN-Zap/zap-desktop/blob/master/ADVANCED.md) page.
 
-### Contributing
-If you would like to help contribute to the project, please see the [contributing guide](https://github.com/LN-Zap/zap-desktop/blob/master/CONTRIBUTING.md).
+If you would like to install from source, run a full bitcoin node, or connect to an custom lnd instance please see the [advanced usage](ADVANCED.md) page.
```
Comment (JimmyMow) [inline README.md]: "...or connect to `a` custom lnd instance..." instead of an

File: README.md [hunk 2]
```diff
@@ -1,27 +1,45 @@
...
+- [Get Help](#get-help)
+- [Maintainers](#maintainers)
+- [Contribute](#contribute)
+- [License](#license)
 
-## Installing
+## Security
 
-***Note:*** *If you would like to use a full bitcoin node, please see the [advanced usage](https://github.com/LN-Zap/zap-desktop/blob/master/ADVANCED.md) page.*
+If you discover or learn about a potential error, weakness, or threat that can compromise the security of Zap, we ask you to keep it confidential and [submit your concern directly to the Zap security team](mailto:security@jackmallers.com?subject=[GitHub]%20Zap%20Security).
```
Comment (JimmyMow) [inline README.md]: I'd prefer `jimmymowschess@gmail.com`. For now at least

**Comment by JimmyMow** [comment]
This is fantastic. Big improvement to the repository. Some super tiny comments from me but the rest looks good

**Comment by mrfelton** [comment]
I've updated those items in the readme @JimmyMow .

I have also added an additional commit which formats the code using the new setup with eslint and prettier - this is required for the tests to pass on travis and to provide a clean base state where everything is formatted correctly before moving forwards. This commit does change a fair amount of code in order to bring consistency across the codebase.

A large amount of these chances come from specific prettier option https://prettier.io/docs/en/options.html#trailing-commas which in our config is set to `all`. This helps improves the clarity of diffs when an item is added or removed from an object or array.

If you could review these updates that would be great.

Happy to discuss/review/adjust any style settings you would like to consider.

**Comment by mrfelton** [comment]
@Empact I have split off the documentation updates and the code linting into the following 2 commits:

https://github.com/LN-Zap/zap-desktop/pull/442
https://github.com/LN-Zap/zap-desktop/pull/443

**Comment by mrfelton** [comment]
Note: the travis build will fail because the code needs to be reformatted.This is handled in https://github.com/LN-Zap/zap-desktop/pull/443

**Comment by Empact** [comment]
@mrfelton Thanks for splitting the docs out! But we def want our PRs and ideally our individual commits to build cleanly, e.g. it can be important for bisecting issues and such, so combining this with #443 is preferred.

**Comment by mrfelton** [comment]
Ok - I've rebased from master and combined those 2 commits - you can review them individually in this PR.

**Comment by Empact** [comment]
Ok so now the next bit is it would be best to match the existing style as much as possible. The main divergence I see is the eslint config was:
`"comma-dangle": ["error", "never"],`

While the prettier config now holds:
`"trailingComma": "all",`

Also I'd like to see something like:
https://github.com/prettier/eslint-config-prettier#no-mixed-operators
Not sure the current state of things but I found this: https://github.com/prettier/prettier/issues/187

Looks like jsx-single-quote is not currently applicable due to maintainer opposition, but I'm fine without that:
https://github.com/prettier/prettier/issues/656
https://github.com/prettier/prettier/pull/3839

**Comment by mrfelton** [comment]
@Empact I've updated those prettier rules to match what we had previously as close as possible in order to keep the changeset of this PR more limited in scope.

I've also broken the commits down further so that it's easier to review the individual changes.

Regarding the trailing comma rule - obviously when it comes to formatting rules it can be very subjective, however with this particular rule there is a good objective reason to always add the trailing commas.

See https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8

I think it would be worth enabling this rule, although we could do that through a separate PR so that change is not included here any more.

**Comment by Empact** [comment]
FYI the trailing commas are still present in the change.

Thanks for being willing to split out changes in our standards for separate consideration btw. Having focused conversation is a great help to development as it allows us to make narrow decisions more quickly.

**Comment by mrfelton** [comment]
> FYI the trailing commas are still present in the change.

Just noticed that - should be resolved now!

**Comment by Empact** [comment]
@mrfelton fyi the changes from #439 do not appear on any open PR afaict, is that intended?

**Comment by mrfelton** [comment]
It's included in the 4th commit on this PR - https://github.com/LN-Zap/zap-desktop/pull/440/commits/4ed944adac0da028a64e6b70888fdaef78349dfa

**Pull request merged**